python-package/xgboost/sklearn.py

```diff
@@ -1481,7 +1537,7 @@ def _cls_predict_proba(n_classes: int, prediction: PredtT, vstack: Callable) ->
         Number of boosting rounds.
 """,
 )
-class XGBClassifier(XGBModel, XGBClassifierBase):
+class XGBClassifier(XGBClassifierBase, XGBModel):
```

Comment:
As of https://github.com/scikit-learn/scikit-learn/pull/30234 (which will be in `scikit-learn` 1.6), the estimator checks raise an error like the following:

> *XGBRegressor is inheriting from mixins in the wrong order. In general, in mixin inheritance, more specialized mixins must come before more general ones. This means, for instance, `BaseEstimator` should be on the right side of most other mixins. You need to change the order...*

That check is new, but it enforced behavior that's been documented in `scikit-learn`'s estimator development docs for a long time. See the "`BaseEstimator` and mixins" section in https://scikit-learn.org/stable/developers/develop.html#rolling-your-own-estimator.

> *It is particularly important to notice that mixins should be “on the left” while the BaseEstimator should be “on the right” in the inheritance list for proper MRO.*

That new check error led to these inheritance-order changes, which led to the `XGBModel.get_params()` changes.